### PR TITLE
[hue] Handle unexpected empty response from API

### LINKED ISSUE #14218: [hue] Exception due to empty response from API (sensors)
Hallo,

To day I updated my OpenHab 3.3 to OpenHab 3.4 and promptly the hue-bridge fails.
I have tried with all combinations of Port (80/443) and Protocol (HTTP/HTTPS), but still always the same result.

In the "events.log" I see the following:
```
023-01-14 12:03:49.543 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:bridge:huebridge' changed from UNINITIALIZED (DISABLED) to INITIALIZING
2023-01-14 12:03:49.550 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:bridge:huebridge' changed from INITIALIZING to UNKNOWN
2023-01-14 12:03:49.554 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:0107:huebridge:PresenceSensor' changed from UNINITIALIZED (BRIDGE_UNINITIALIZED) to INITIALIZING
2023-01-14 12:03:49.556 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:0010:huebridge:3' changed from UNINITIALIZED (BRIDGE_UNINITIALIZED) to INITIALIZING
2023-01-14 12:03:49.560 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:0010:huebridge:2' changed from UNINITIALIZED (BRIDGE_UNINITIALIZED) to INITIALIZING
2023-01-14 12:03:49.561 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:0010:huebridge:3' changed from INITIALIZING to OFFLINE (BRIDGE_OFFLINE)
2023-01-14 12:03:49.567 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:0107:huebridge:PresenceSensor' changed from INITIALIZING to OFFLINE (BRIDGE_OFFLINE)
2023-01-14 12:03:49.569 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:0010:huebridge:2' changed from INITIALIZING to OFFLINE (BRIDGE_OFFLINE)
2023-01-14 12:03:52.571 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:bridge:huebridge' changed from UNKNOWN to ONLINE
2023-01-14 12:03:52.573 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:0107:huebridge:PresenceSensor' changed from OFFLINE (BRIDGE_OFFLINE) to ONLINE
2023-01-14 12:03:52.575 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:0010:huebridge:2' changed from OFFLINE (BRIDGE_OFFLINE) to ONLINE
2023-01-14 12:03:52.576 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:0010:huebridge:3' changed from OFFLINE (BRIDGE_OFFLINE) to ONLINE
2023-01-14 12:03:54.130 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:bridge:huebridge' changed from ONLINE to OFFLINE: Hue Bridge connection lost.
2023-01-14 12:03:54.131 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:0010:huebridge:2' changed from ONLINE to OFFLINE (BRIDGE_OFFLINE)
2023-01-14 12:03:54.132 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:0010:huebridge:3' changed from ONLINE to OFFLINE (BRIDGE_OFFLINE)
2023-01-14 12:03:54.133 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:0107:huebridge:PresenceSensor' changed from ONLINE to OFFLINE (BRIDGE_OFFLINE)
2023-01-14 12:03:54.566 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:bridge:huebridge' changed from OFFLINE: Hue Bridge connection lost. to ONLINE
2023-01-14 12:03:54.568 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:0010:huebridge:2' changed from OFFLINE (BRIDGE_OFFLINE) to ONLINE
2023-01-14 12:03:54.569 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:0010:huebridge:3' changed from OFFLINE (BRIDGE_OFFLINE) to ONLINE
2023-01-14 12:03:54.570 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:0107:huebridge:PresenceSensor' changed from OFFLINE (BRIDGE_OFFLINE) to ONLINE
2023-01-14 12:03:54.637 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:bridge:huebridge' changed from ONLINE to OFFLINE: Hue Bridge connection lost.
2023-01-14 12:03:54.638 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:0010:huebridge:2' changed from ONLINE to OFFLINE (BRIDGE_OFFLINE)
2023-01-14 12:03:54.638 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:0010:huebridge:3' changed from ONLINE to OFFLINE (BRIDGE_OFFLINE)
2023-01-14 12:03:54.639 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:0107:huebridge:PresenceSensor' changed from ONLINE to OFFLINE (BRIDGE_OFFLINE)
2023-01-14 12:03:58.193 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:bridge:huebridge' changed from OFFLINE: Hue Bridge connection lost. to ONLINE
2023-01-14 12:03:58.194 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:0010:huebridge:3' changed from OFFLINE (BRIDGE_OFFLINE) to ONLINE
2023-01-14 12:03:58.195 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:0107:huebridge:PresenceSensor' changed from OFFLINE (BRIDGE_OFFLINE) to ONLINE
2023-01-14 12:03:58.196 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:0010:huebridge:2' changed from OFFLINE (BRIDGE_OFFLINE) to ONLINE
2023-01-14 12:03:58.698 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:bridge:huebridge' changed from ONLINE to OFFLINE: Hue Bridge connection lost.
2023-01-14 12:03:58.699 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:0010:huebridge:3' changed from ONLINE to OFFLINE (BRIDGE_OFFLINE)
2023-01-14 12:03:58.700 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:0010:huebridge:2' changed from ONLINE to OFFLINE (BRIDGE_OFFLINE)
2023-01-14 12:03:58.700 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'hue:0107:huebridge:PresenceSensor' changed from ONLINE to OFFLINE (BRIDGE_OFFLINE)

```
And in openhab.log the following repeats over and over again:
```
2023-01-14 12:03:58.696 [WARN ] [ue.internal.handler.HueBridgeHandler] - An unexpected error occurred: null
java.lang.NullPointerException: null
        at org.openhab.binding.hue.internal.connection.HueBridge.getSensors(HueBridge.java:280) ~[bundleFile:?]
        at org.openhab.binding.hue.internal.handler.HueBridgeHandler$1.doConnectedRun(HueBridgeHandler.java:199) ~[bundleFile:?]
        at org.openhab.binding.hue.internal.handler.HueBridgeHandler$PollingRunnable.run(HueBridgeHandler.java:136) [bundleFile:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305) [?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:829) [?:?]

```

## Expected Behavior
hue-bridge starts as "usual" :-)

## Your Environment
* Version used: openHAB 3.4.1, hue-binding 3.4.1
* Operating System and version:
  OpenHab: Fedora Linux 35 (Server Edition)
  deconz: Raspbian GNU/Linux 10 (buster), deconz version 2.19.03 / 9/19/2022

Any suggestions?

Dan

### PR DESCRIPTION
**Pull request by lolodomo**
Fix #14218

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
In the issue, we discovered that the API can sometimes return an empty response when requesting the sensors data.
This PR allows ignoring empty response when lights/groups/scenes are requested by the scheduled poll job.

File: bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/exceptions/EmptyResponseException.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+package org.openhab.binding.hue.internal.exceptions;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * Thrown when API is returning an unexpected empty response.
+ *
+ * @author Laurent Garnier - Initial contribution
+ */
+@SuppressWarnings("serial")
```
Comment (jlaur) [inline bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/exceptions/EmptyResponseException.java:22]: I would suggest to remove this suppression and instead add:
```java
private static final long serialVersionUID = 1L;
```

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks, LGTM. Only one minor comment.

File: bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/exceptions/EmptyResponseException.java (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/exceptions/EmptyResponseException.java:22]: I copy/paste another exception of this binding which are all using @SuppressWarnings("serial"). 
I keep it on the new class to be consistent with others.

**Review by jlaur — APPROVED** [review]
LGTM

**Comment by lolodomo** [comment]
Maybe a candidate for 3.4.x branch?

**Comment by jlaur** [comment]
> Maybe a candidate for 3.4.x branch?

Agreed, I'll cherry-pick in the evening.

**Pull request merged**